internal/checker/checker.go

```diff
@@ -15389,8 +15403,33 @@ func (c *Checker) widenTypeInferredFromInitializer(declaration *ast.Node, t *Typ
 
 func (c *Checker) getTypeOfFuncClassEnumModule(symbol *ast.Symbol) *Type {
 	links := c.valueSymbolLinks.Get(symbol)
+	originalLinks := links
 	if links.resolvedType == nil {
+		expando := c.getSymbolOfExpando(symbol.ValueDeclaration)
+		if expando != nil {
```

Comment:
this ports Strada's `mergeJSSymbols` - but since JS is not really involved here right now, I decided to just inline its relevant parts here (at least for the time being)